File: Web_API/Web_API/Models/Restaurant.cs
using System;
using System.Collections.Generic;

namespace Web_API.Models
{
	public class Restaurant
	{
		public int Id { get; set; }
		public string Name { get; set; }
		public string Address { get; set; }
		public double Latitude { get; set; } // Latitude property
		public double Longitude { get; set; } // Longitude property
		public ICollection<Product> Products { get; set; }
	}
}

File: assignment/lib/models/restaurant.dart
import 'package:assignment/models/product.dart';

class Restaurant {
  final int id;
  final String name;
  final String address;
  final double latitude;
  final double longitude;

  Restaurant({
    required this.id,
    required this.name,
    required this.address,
    required this.latitude,
    required this.longitude,
  });

  factory Restaurant.fromJson(Map<String, dynamic> json) {
    return Restaurant(
      id: json['Id'],
      name: json['Name'],
      address: json['Address'],
      latitude: json['Latitude'],
      longitude: json['Longitude'],
    );
  }
}
